install.sh

```diff
@@ -0,0 +1,345 @@
+    exit 1
+  fi
+
+  distro_name
+
+  ARCH="$(arch)"
+  if [ ! "$ARCH" ]; then
+    if [ "${STATIC-}" ]; then
+      echo "No static releases available for the architecture $(uname -m)."
+      echo "Please rerun without the -s flag to install from npm."
+      exit 1
+    fi
+    install_npm
+    exit 0
+  fi
+
+  CACHE_DIR="$(cache_dir)"
+  mkdir -p "$CACHE_DIR"
+
+  if [ "${STATIC-}" ]; then
+    install_static
+    return
+  fi
+
+  case "$(distro)" in
+  macos)
+    install_macos
+    ;;
+  ubuntu | debian | raspbian)
+    install_deb
+    ;;
+  centos | fedora | rhel | opensuse)
+    install_rpm
+    ;;
+  arch)
+    install_arch
+    ;;
+  *)
+    install_static
+    ;;
+
+  esac
+}
+
+install_macos() {
+  if command_exists brew; then
+    echo "Installing from Homebrew."
+
+    sh_c brew install code-server
+
+    return
+  fi
+
+  echo "Homebrew is not installed so installing static release."
+
+  install_static
+}
+
+fetch() {
+  URL="$1"
+  FILE="$2"
+
+  if [ -e "$FILE" ]; then
+    echo
+    echo "+ Using cached $FILE from $URL"
+    return
+  fi
+
+  sh_c curl \
+    -#fL \
+    -Ro "$FILE.incomplete" \
+    -C - \
+    "$URL"
+  mv "$FILE.incomplete" "$FILE"
+}
+
+install_deb() {
+  echo "Installing v$VERSION deb package from GitHub releases."
+
+  fetch "https://github.com/cdr/code-server/releases/download/v$VERSION/code-server_${VERSION}_$ARCH.deb" "$CACHE_DIR/code-server_${VERSION}_$ARCH.deb"
+  sudo_sh_c dpkg -i "$CACHE_DIR/code-server_${VERSION}_$ARCH.deb"
+
+  echo_systemd_postinstall
+}
+
+install_rpm() {
+  echo "Installing v$VERSION rpm package from GitHub releases."
+
+  fetch "https://github.com/cdr/code-server/releases/download/v$VERSION/code-server-$VERSION-$ARCH.rpm" "$CACHE_DIR/code-server-$VERSION-$ARCH.rpm"
+  sudo_sh_c rpm -i "$CACHE_DIR/code-server-$VERSION-$ARCH.rpm"
+
+  echo_systemd_postinstall
+}
+
+install_arch() {
+  echo "Installing from the AUR."
+
+  fetch "https://aur.archlinux.org/cgit/aur.git/snapshot/code-server.tar.gz" "$CACHE_DIR/code-server-aur.tar.gz"
+
+  tmp_dir="$(mktemp -d)"
+  (
+    cd "$tmp_dir"
+    tar -xzf "$CACHE_DIR/code-server-aur.tar.gz" --strip-components 1
+    sh_c makepkg -si
+  )
+  rm -Rf "$tmp_dir"
+
+  echo_systemd_postinstall
+}
+
+install_static() {
+  STATIC_INSTALL_PREFIX=${STATIC_INSTALL_PREFIX-/usr/local/lib}
+
+  echo "Installing static release v$VERSION"
+
+  fetch "https://github.com/cdr/code-server/releases/download/v$VERSION/code-server-$VERSION-$OS-$ARCH.tar.gz" "$CACHE_DIR/code-server-$VERSION-$OS-$ARCH.tar.gz"
+
+  if [ ! -d "$STATIC_INSTALL_PREFIX" ]; then
+    echo
+    echo "Static release install prefix $STATIC_INSTALL_PREFIX does not exist"
+    exit 1
+  fi
+
+  sh_c="sh_c"
+  if [ ! -w "$STATIC_INSTALL_PREFIX" ]; then
+    sh_c="sudo_sh_c"
+  fi
+
+  "$sh_c" tar -C "$STATIC_INSTALL_PREFIX" -xzf "$CACHE_DIR/code-server-$VERSION-$OS-$ARCH.tar.gz"
+  # In case previously installed.
+  SKIP_LOG=1 "$sh_c" rm -Rf "$STATIC_INSTALL_PREFIX/code-server-$VERSION"
```

Comment:
Should we confirm this or do a `mv` instead? I feel like it's probably not going to bite anyone and this is probably how most or all other install scripts operate; I'm just thinking about the issue we had with that user storing their data in the code-server dir. That was probably a symptom of manually installing though and so might not be worth considering here.

Reply:
Yea will exit out just in case.